Add TemplateInput models and form

ares opened the pull request:
based on #1

ehelms commented on lib/foreman_remote_execution/engine.rb:
```diff
@@ -0,0 +1,93 @@
+        {:assets => {:precompile => [
+          'template_input.js',
+        ]}}
+    end
```
Do you need both this declaration for generating plugin assets and the declaration on line 14?

ehelms commented on lib/foreman_remote_execution/engine.rb:
```diff
@@ -0,0 +1,93 @@
+    end
+
+    rake_tasks do
+      Rake::Task['db:seed'].enhance do
```
Consider taking advantage of using `seeds.d` (https://github.com/theforeman/foreman/blob/develop/db/seeds.rb#L53)

iNecas replied:
I think we can just remove this line until the seeds are needed

ares commented:
@ehelms thanks for reviews, could you continue in #1? This PR should add just the last commit, but I didn't want to wait until it's merged... Anyway I based this on foreman plugin template so if you think some of these should be changed, it would be a good idea to modify the template as well.

ehelms commented:
Good idea. I'll start with updates to the template.

jlsherrill commented on app/models/template_input.rb:
```diff
@@ -0,0 +1,10 @@
+  belongs_to :job_template
+
+  validates :name, :presence => true
+  validates :input_type, :presence => true
```
maybe add an inclusion in TYPES validation?

jlsherrill commented:
I can't seem to create a Job template with this change.  It always flashes to the 'Template Input' section and says that "Name can't be blank".  (regardless of whether i've added an intput or not, if i have added an input it 'adds' another empty input section and complains about that).

Also for some reason when creating a new template the input type is 'stuck' with User Input.  I can't seem to change it.

ares commented:
argh sorry I forgot to mention we need this https://github.com/theforeman/foreman/pull/2468 in Foreman (not sure if it also solves the second issue)

jlsherrill commented:
aaaaahhh, yes that seemed to fix the first issue, but not the 'locked' selection on input type.  It seems to only happen for 'new inputs' on the job create & edit screen.  If i create the job with an input of 'user input' type, i can then change it later on.

ares commented:
I moved the validation from my WIP branch to this commit. Also I've added 4th input type since we have to distinguish between smart class parameters and global parameters (will be merged with smart variables). The first one needs specifying a smart class name. Also added more validations and fixed JS to display correct fields on validation errors.

The issue with select you found was introduced in Foreman nightly (they merged select2 PR) so I had to create another fix in Foreman, please test with https://github.com/theforeman/foreman/pull/2482

Good news is that theforeman/foreman#2468 got in develop already so you just need to update your Foreman.

Could you give it another rounds please?

jlsherrill commented:
All looks good to me, ACK

ares commented:
Thanks for the review, merging!
